scipy/spatial/_hausdorff.pyx

```diff
@@ -0,0 +1,63 @@
+"""
+Directed Hausdorff Code
+
+.. versionadded:: 0.19.0
+
+"""
+#
+# Copyright (C)  Tyler Reddy and Richard Gowers, 2016
+#
+# Distributed under the same BSD license as Scipy.
+#
+
+import numpy as np
+cimport numpy as np
+cimport cython
+from libc.math cimport sqrt
+
+__all__ = ['directed_hausdorff']
+
+@cython.boundscheck(False)
+def directed_hausdorff(double[:,::1] ar1, double[:,::1] ar2):
+
+    cdef double cmax, cmin, d
+    cdef int break_occurred
+    cdef int N1 = ar1.shape[0]
+    cdef int N2 = ar2.shape[0]
+    cdef int data_dims = ar1.shape[1]
+    cdef unsigned int i, j, k, i_store, j_store, i_ret, j_ret
```

Comment:
To prefer signed types is just a friendly hint. Code will work also with the unsigned type but it can have hard to find bugs due to overflow. But your arrays can only be as long as an int since you declared `N1, N2` to be ints. To be consistent it would be nice if they all share the same type, this includes the type of `resort1, resort2`.